src/Publish/Microsoft.NETCore.Sdk.Publish.Tasks/Tasks/TransformWebConfig.cs

```diff
@@ -74,6 +82,11 @@ public override bool Execute()
             string outputFile = Path.GetFileName(TargetPath);
             XDocument transformedConfig = WebConfigTransform.Transform(webConfigXml, outputFile, IsAzure, IsPortable);
 
+            if (string.IsNullOrEmpty(ProjectGuid) && !IgnoreProjectGuid)
+            {
+                ProjectGuid = WebConfigTransform.GetProjectGuidFromSolutionFile(SolutionPath, ProjectFullPath);
+            }
+
             // Add the projectGuid to web.config if it is not present.
             transformedConfig = WebConfigTransform.AddProjectGuidToWebConfig(transformedConfig, ProjectGuid, IgnoreProjectGuid);
```

Comment:
do you mean the CLI opt out variable (DOTNET_CLI_TELEMETRY_OPTOUT)? I can add a check to make sure that if this is set, then "dotnet publish" wont track the projectGuid.